src/plugins/vis_type_table_new/public/components/table_vis_component.tsx

```diff
@@ -38,8 +38,8 @@ export const TableVisComponent = ({
   const pagination = usePagination(visConfig, rows.length);
 
   const sortedRows = useMemo(() => {
-    return uiState.sort?.colIndex && uiState.sort.direction
-      ? orderBy(rows, columns[uiState.sort.colIndex]?.id, uiState.sort.direction)
+    return uiState.sort.colIndex !== null && uiState.sort.direction
+      ? orderBy(rows, columns[uiState.sort.colIndex].id, uiState.sort.direction)
       : rows;
```

Comment:
I am still not comfortable that we are not checking the validity of `id` before passing it to `orderBy`. 


PS, I would have preferred for lodash not even be used here.

Reply:
So, from the description, it sounds like the bug was that `colIndex === 0` evaluates falsey, which was not what you intended to check. 

But I agree with @AMoo-Miki - what about the case where `columns` is an empty array, or where the `colIndex` is out of bounds? Seems like you also want to check `columns[uiState.sort.colIndex].id` in your conditional.

Reply:
@joshuarrrr From what I know, if there is no data, there are two cases:
* for `count` metric, table rows 
```
0: 
{col-0-1: 0}
```
* for other metric, table rows:
```
0: 
{col-0-1: null}
```

So no matter what metric we use, we could sort. There is always a column. But if we think this is not 100% percent guaranteed, we could add more check 

```
uiState.sort.colIndex !== null && columns[uiState.sort.colIndex].id && uiState.sort.direction
```

Reply:
I think `uiState.sort.colIndex !== null && columns[uiState.sort.colIndex].id && uiState.sort.direction` would be the best.

I understand that there are simpler and more effective checks and TS is being stupid.

Reply:
@AMoo-Miki agree. I committed the change.